internal/blockdb/migrate.go

```diff
@@ -80,5 +80,27 @@ ON CONFLICT(git_sha) DO UPDATE SET git_sha=git_sha`, nowRFC3339(), gitSha)
 		return fmt.Errorf("create table tx: %w", err)
 	}
 
+	_, err = db.Exec(`CREATE VIEW IF NOT EXISTS tx_flattened AS
```

Comment:
A few comments. To be clear, I love this idea!

## Idempotent

To keep the migrations idempotent, I recommend [dropping the view if exists](https://www.sqlitetutorial.net/sqlite-drop-view/) first. A view is essentialy a stored query so should be fast to drop/re-create. It will allow us to change the views more easily when we change table columns.

Also, we probably want to keep view creation as the very last migration steps (as you've done here). 

## Naming Convention

I like prefixing views with `v_` to signal it's a view a not a table. E.g. `v_tx_flattened`. 

## Delcare Columns Separately

This one is a nit for style.

It may read nicer to declare the columns separately as shown [here](https://www.sqlitetutorial.net/sqlite-create-view/). Look at the section `2) Creating a view with custom column names`. This is instead of using `SELECT col as another_name`. 

E.g. (per the article)

```
CREATE VIEW v_albums (
    AlbumTitle,
    Minutes
)
AS
    SELECT albums.title,
           SUM(milliseconds) / 60000
      FROM tracks
```

Reply:
> It may read nicer to declare the columns separately

That is likely true in cases where there are a small number of columns in the view, but IMO with ten columns in `v_tx_flattened`, it is more explicit to use `AS` to name each column; less chance of someone reordering the columns and creating a mismatch. Although, at the same time, we do have tests against this view.

@agouin would you care to be a tiebreaker on column separation in the built view? https://github.com/strangelove-ventures/ibctest/blob/29d64faf404be43f6e2512f6c690f38c2a6f8044/internal/blockdb/migrate.go#L88-L103